web/src/rendered_markdown.ts

```diff
@@ -192,38 +204,44 @@ export const update_elements = ($content) => {
         }
     });
 
-    $content.find("a.stream").each(function () {
-        const stream_id = Number.parseInt($(this).attr("data-stream-id"), 10);
-        if (stream_id && !$(this).find(".highlight").length) {
-            // Display the current name for stream if it is not
-            // being displayed in search highlight.
-            const stream_name = sub_store.maybe_get_stream_name(stream_id);
-            if (stream_name !== undefined) {
-                // If the stream has been deleted,
-                // sub_store.maybe_get_stream_name might return
-                // undefined.  Otherwise, display the current stream name.
-                $(this).text("#" + stream_name);
+    $content.find("a.stream").each(function (): void {
+        const stream_id_string = $(this).attr("data-stream-id");
+        if (stream_id_string !== undefined) {
+            const stream_id = Number.parseInt(stream_id_string, 10);
+            if (stream_id && !$(this).find(".highlight").length) {
```

Comment:
As with user groups, I confirmed in 1d1c41551bc4168765ab5c81c8fbcab10053c8a9 (the original stream mention commit) that we can expect `data-stream-id` to be present and an integer for a correct server. So it could be reasonable to be more aggressive with assertions here rather than having all these if/do-nothing clauses.

@andersk thoughts?

Reply:
Sure. Eventually we will want this code to become secure against malicious servers, but rejecting invalid input with an error is a fine way to fail securely, and establishing this invariant will help us understand the expected valid input when we refactor this later.

Reply:
Sure, changed to asserting that the stream id is defined